testing/bats/root-objects.bats

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/setup/common.bash
+
+setup() {
+    setup_common
+    start_sql_server
+}
+
+teardown() {
+    teardown_common
+}
+
+@test 'root-objects: dolt_add, dolt_branch, dolt_checkout, dolt_commit, dolt_reset' {
```

Comment:
Probably want to also stop and restart the server in this test and make sure everything survives.

Also, would be nice to have a broader selection of root objects, not just sequences.

Also also: do we have any go tests that catch the problems you discovered about things being cached in the session and not actually being persisted? We should.

Reply:
Added a test for starting and stopping. Only chose to do sequences for the bats test since this is mainly testing the root object path, which is common for all the types. There will be differences between the types that manifest in other ways (merging, etc.) but those _theoretically_ will be caught by engine tests. Also, I do have Go tests for the session persistence issue.